src/dotnet-install.ps1

```diff
@@ -204,6 +204,15 @@ function Get-Machine-Architecture() {
         return $ENV:PROCESSOR_ARCHITEW6432
     }
 
+    # covers the case when PS x64 is run on ARM machine
+    if( ((Get-WmiObject Win32_OperatingSystem).OSArchitecture) -like "ARM*") {
```

Comment:
`Get-WmiObject` also seems [outdated](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/get-wmiobject?view=powershell-5.1#description). The following should be the way if I'm not mistaken:
```suggestion
    if( ((Get-CimInstance -ClassName CIM_OperatingSystem).OSArchitecture) -like "ARM*") {
```